src/asm/rgbasm.5

```diff
@@ -0,0 +1,279 @@
+Mapped to memory at $FE00-$FE9F.
+Can only allocate memory, not fill it.
+.It Sy HRAM
+A high RAM section.
+Mapped to memory at $FF80–$FFFE.
+Can only allocate memory, not fill it.
+.Pp
+NOTE: If you use this method of allocating HRAM the assembler will NOT choose
+the short addressing mode in the LD instructions
+.Sy LD [$FF00+n8],A No and Sy LD A,[$FF00+n8] No because the actual address
+calculation is done by the linker.
+If you find this undesirable you can use
+.Ar RSSET No / Ar RB No / Ar RW No instead or use the
+.Sy LDH [$FF00+n8],A No and Sy LDH A,[$FF00+n8] No syntax instead.
+This forces the assembler to emmit the correct instruction and the linker to
+check if the value is in the correct range.
+.El
+.Pp
+A section is usually defined as a floating one, but the code can restrict where
+the linker can place it.
+.Pp
+If a section is defined with no indications, it is a floating section.
+The linker will decide where to place it in the final binary and it has no
+obligation to follow any specific rules.
+The following example defines a section that can be placed anywhere in any ROMX
+bank:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX
+.Pp
+If it is needed, the following syntax can be used to fix the base address of the
+section:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX[$4567]
+.Pp
+It won't, however, fix the bank number, which is left to the linker.
+If you also want to specify the bank you can do:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX[$4567],BANK[3]
+.Pp
+And if you only want to force the section into a certain bank, and not it's
+position within the bank, that's also possible:
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX,BANK[7]
+.Pp
+In addition, you can specify byte alignment for a section.
+This ensures that the section starts at a memory address where the given number
+of least-significant bits are 0.
+This can be used along with
+.Ar BANK No , if desired.
+However, if an alignment is specified, the base address must be left unassigned.
+This can be useful when using DMA to copy data or when it is needed to align the
+start of an array to 256 bytes to optimize the code that accesses it.
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX,ALIGN[8] ; align to 256 bytes
+.Pp
+.Dl SECTION \*qCoolStuff\*q,ROMX,BANK[2],ALIGN[4] ; align to 16 bytes
+.Pp
+HINT: If you think this is a lot of typing for doing a simple
+.Ar ORG No type thing you can quite easily write an intelligent macro (called
+.Ar ORG No for example) that uses Ar \@ No for the section name and determines
+correct section type etc as arguments for
+.Ar SECTION .
+.Pp
+Sections can also be placed by using a linkerscript file.
+The format is described in
+.Xr rgblink 5 .
+They allow the user to place floating sections in the desired bank in the order
+specified in the script.
+This is useful if the sections can't be placed at an address manually because
+the size may change, but they have to be together.
+.Pp
+.Sh SYMBOLS
+.Pp
+.Ss Symbols
+
+
+
+
+
+
+
+.Sh ALPHABETICAL LIST OF KEYWORDS
```

Comment:
First I'd like to port all the documentation, we can improve it later. If I have time to do it before Monday I'll do it. My idea is to create a new release this weekend, so I'm not going to do commit anything more that isn't a bugfix (or documentation).